### PR TITLE
AEROGEAR-2929 Update the navigation menu

### PR DESCRIPTION
**Pull request by aidenkeating**
Currently the navigation menu does not have the features required.

These features are:
* An updated navigation structure with icons for parents and indented children.
* Updated background color.
* Updated navigation header icon - More padding.
* Updated text beside icon - AeroGear aerogear.org.
* Screen titles that reflect the current page.
* An under construction page for metrics.

There is also logic included for opening the documentation pages once they are
available, and also including the metrics sub-sections in the navigation menu
once they have been added.

### CODE REVIEW

File: ios-showcase-template/root/RootViewController.swift [hunk 1]
```diff
@@ -7,29 +7,100 @@
...
+    
+    init(_ type: MenuItemType, _ title: String, _ iconName: String?) {
+        self.type = type
+        self.title = title
+        self.iconName = iconName
+    }
+}
+
+enum MenuItemType {
+    case Home
```
Comment (aidenkeating) [inline ios-showcase-template/root/RootViewController.swift]: These should be camelCase

File: ios-showcase-template/menu/MenuViewController.swift [hunk 2]
```diff
@@ -97,7 +97,7 @@ class MenuViewController: UIViewController, UITableViewDataSource, UITableViewDe
     */
 
     func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
-        let cell: UITableViewCell = tableView.dequeueReusableCell(withIdentifier: "cellMenu")!
+        let cell: UITableViewCell = tableView.dequeueReusableCell(withIdentifier: "cellMenu", for: indexPath)
```
Comment (aidenkeating) [inline ios-showcase-template/menu/MenuViewController.swift]: This addition isn't needed

File: ios-showcase-template/root/RootViewController.swift [hunk 3]
```diff
@@ -7,29 +7,100 @@
...
+        // MenuItem(MenuItemType.MetricsDocs, MENU_ITEM_TITLE_METRICS_DOCS, nil),
+        // MenuItem(MenuItemType.MetricsProfile, MENU_ITEM_TITLE_METRICS_PROFILE, nil),
+        // MenuItem(MenuItemType.MetricsTrust, MENU_ITEM_TITLE_METRICS_TRUST, nil)
+    ]
 
     var listener: MenuListener?
 
     override func viewDidLoad() {
         super.viewDidLoad()
+
```
Comment (aidenkeating) [inline ios-showcase-template/root/RootViewController.swift]: Not needed.

**Comment by aidenkeating** [comment]
@wei-lee Would you mind taking a look? I have included screenshots below of the current progress.

screen shot 2018-06-11 at 16 55 51
screen shot 2018-06-11 at 16 56 02

**Comment by wei-lee** [comment]
so far looks good. The only comment I have is the link should be clickable in the under construction page.

**Review by wei-lee — APPROVED** [review]

**Comment by aidenkeating** [comment]
@StephenCoady @secondsun @ziccardi  Mind taking a look also

**Review by StephenCoady — APPROVED** [review]
looks good. looking at this and my cordova changes they are in line.

**Comment by aidenkeating** [comment]
screen shot 2018-06-12 at 11 26 04

screen shot 2018-06-12 at 11 04 48

**Comment by aidenkeating** [comment]
@StephenCoady @wei-lee  Mind taking a look? I've updated the under construction page slightly, just splitting the paragraph really. Also some home page updates

**Review by wei-lee — APPROVED** [review]

**Review by StephenCoady — APPROVED** [review]

**Comment by aidenkeating** [comment]
@wei-lee Mind taking one more quick look/verification?

* home screen - https://docs.google.com/presentation/d/1FZL-Wy5D_wUo5ADrfhpRBh-Ny2rGauI3_C8SBrlk6rw/edit#slide=id.g3c52d834c7_0_30
* nav menu - https://docs.google.com/presentation/d/1FZL-Wy5D_wUo5ADrfhpRBh-Ny2rGauI3_C8SBrlk6rw/edit#slide=id.g3c52d834c7_0_44
* under construction/metrics - https://docs.google.com/presentation/d/1FZL-Wy5D_wUo5ADrfhpRBh-Ny2rGauI3_C8SBrlk6rw/edit#slide=id.g3c52d834c7_0_128

**Comment by wei-lee** [comment]
@aidenkeating the only comment I have is can we change the color of the fonts in the nav menu to be the same as the cordova one (should be black I think).

The other problem is the way menu is presented when it is open. But we can discuss it during standup.

**Comment by aidenkeating** [comment]
@wei-lee I see the issue you're talking about on the menu, I think I have fixed it (updated the slideshow). Is that what you were talking about?

**Comment by wei-lee** [comment]
@aidenkeating yes, almost there. One small thing is about the separators. On the ios one it's from end to end, but the cordova one is not. We can discuss it during the standup which one is easier to implement and go with that approach.

**Comment by aidenkeating** [comment]
@wei-lee Updated the nav menu dividers to not be all the way across, made the font color black and updated the header color to match the rest of the app

**Comment by wei-lee** [comment]
👍

**Review by wei-lee — APPROVED** [review]

**Pull request merged**